pkg/nuctl/test/functionevent_test.go

```diff
@@ -39,20 +42,34 @@ func (suite *functionEventGetTestSuite) TestGet() {
 	for functionEventIdx := 0; functionEventIdx < numOfFunctionEvents; functionEventIdx++ {
 		uniqueSuffix := fmt.Sprintf("-%s-%d", xid.New().String(), functionEventIdx)
 
+		runtimeName, _ := common.GetRuntimeNameAndVersion("shell")
+		functionName := fmt.Sprintf("function-%d", functionEventIdx)
+		namedArgs := map[string]string{
+			"path":    path.Join(suite.GetExamples(), runtimeName, "empty", "empty.sh"),
+			"runtime": "shell",
+			"handler": "empty.sh:main",
+		}
+		suite.logger.DebugWith("Deploying function",
```

Comment:
did you add a validation that required you to change the test to deploy a function first?

Reply:
Yes, if you add a function event which belongs to a function, the platform needs to check its OPA permissions on the resource `/projects/<project_name>/functions/<function_name>/function_events/<event_name>`.
To get the project name, we need to get the function object, and for that to happen the function needs to exist.